File: frontend/src/requests/problem_set_generator.js
 // API response is of the form:
 //
 //     {'quantity':{0:2, 1:1},
 //      'correct': {0:1, 1:0},
 //      'grades': {0:.92, 1:.5},
 //      'graded': False,
 //      'problems': [
 //          {'question': '', 'answer':'', 'difficulty': 0, 'result':True},
 //          {'question': '', 'answer':'', 'difficulty': 0, 'result':False},
 //          {'question': '', 'answer':'', 'difficulty': 1, 'result':False}
 //          ],
 //      }


fetchProblemSet() {
  fetch("http://localhost:5000/api/problem_set_generator/10/2")
    .then(response => response.json())
    .then((data) => {this.setState({
      problem_set: data,
      question: data['problems'][this.counter]['question'],
      correct_answer: data['problems'][this.counter]['answer']
    })})
    .catch(error => {
      console.log(error);
      this.setState({
        error: error
      })
    });
}

File: learncentive/api/problem_set_generator.py
from random import randint
import json
from flask_restful import Resource

from learncentive.problem_generation.problem_set import ProblemSet

class ProblemSetGenerator(Resource):
    """ API response is of the form:
       {'grades': [.92, .5]
        'graded': False,
        'problems': [
            {'question': '', 'answer':'', 'difficulty': 0, 'result':True},
            {'question': '', 'answer':'', 'difficulty': 0, 'result':False},
            {'question': '', 'answer':'', 'difficulty': 1, 'result':False}
            ],
        }
    """
    def get(self, json_data):
        old_problem_set = ProblemSet(json.loads(json_data))
        new_problem_set = old_problem_set.new_problem_set()
        return json.dumps(vars(new_problem_set))

class ProblemSetALaCarte(Resource):

    def get(self, amount_of_probs, type_of_prob):
        new_problem_set = ProblemSet.alacarte(amount_of_probs, type_of_prob)
        return json.dumps(vars(new_problem_set))
